src/custom/state/claim/hooks/index.ts

```diff
@@ -1000,3 +1000,70 @@ export const useClaimLinks = () => {
     [chainId]
   )
 }
+
+/**
+ * Hook that parses the result input with BigNumber value to CurrencyAmount
+ */
+function useParseVCowResult(result: Result | undefined) {
+  const { chainId } = useActiveWeb3React()
+
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  return useMemo(() => {
+    if (!chainId || !vCowToken || !result) {
+      return
+    }
+
+    return CurrencyAmount.fromRawAmount(vCowToken, result[0].toString())
+  }, [chainId, result, vCowToken])
+}
+
+/**
+ * Hook that fetches the needed vCow data and returns it in VCowData type
+ */
+type VCowData = {
+  isLoading: boolean
+  total: CurrencyAmount<Currency> | undefined | null
+  unvested: CurrencyAmount<Currency> | undefined | null
+  vested: CurrencyAmount<Currency> | undefined | null
+}
+
+export function useVCowData(): VCowData {
+  const vCowContract = useVCowContract()
+  const { account } = useActiveWeb3React()
+
+  const singleCallInput = useMemo(() => [account ?? undefined], [account])
+
+  const { loading: isVestedLoading, result: vestedResult } = useSingleCallResult(
+    vCowContract,
+    'swappableBalanceOf',
+    singleCallInput
+  )
+  const { loading: isTotalLoading, result: totalResult } = useSingleCallResult(
+    vCowContract,
+    'balanceOf',
+    singleCallInput
+  )
+
+  const vested = useParseVCowResult(vestedResult)
+  const total = useParseVCowResult(totalResult)
+
+  const unvested = useMemo(() => {
+    if (!total || !vested) {
+      return null
+    }
+
+    // Check if total < vested, if it is something is probably wrong and we return null
+    if (total.lessThan(vested)) {
+      return null
+    }
+
+    return total?.subtract(vested)
```

Comment:
I believe it's not possible to be undefined by this point:

```suggestion
    return total.subtract(vested)
```

Reply:
Updated :+1: